### PR TITLE
drivers: Merge the common driver initialization into one place

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Impact
Minor, code refactor

## Testing
Pass CI

### CODE REVIEW

File: arch/sim/src/sim/up_initialize.c [hunk 1]
```diff
@@ -351,5 +243,7 @@ void up_initialize(void)
   audio_register("pcm0c", sim_audio_initialize(false));
 #endif
 
-  up_loop_init();
+  kthread_create("loop_task", 1,
```
Comment (pkarashchenko) [inline arch/sim/src/sim/up_initialize.c:246]:
```suggestion
  kthread_create("loop_task", SCHED_PRIORITY_MIN,
```

File: arch/mips/src/common/mips_initialize.c [hunk 2]
```diff
@@ -67,10 +52,6 @@
 
 void up_initialize(void)
 {
-  /* Initialize global variables */
-
-  CURRENT_REGS = NULL;
```
Comment (pkarashchenko) [inline arch/mips/src/common/mips_initialize.c:72]: Just to make sure that this is intentional change

Comment (xiaoxiang781216) [reply arch/mips/src/common/mips_initialize.c:72]: Yes, the global variable is already zero.

File: arch/sim/src/sim/up_initialize.c (hunk 1 above)
Comment (xiaoxiang781216) [reply arch/sim/src/sim/up_initialize.c]: Done.

**Review by pkarashchenko — APPROVED** [review]

**Comment by masayuki2009** [comment]
@xiaoxiang781216 
I noticed that qemu-intel64 does not work correctly after this PR is merged.

**Comment by xiaoxiang781216** [comment]
@masayuki2009 I follow https://github.com/apache/incubator-nuttx/blob/master/boards/x86_64/intel64/qemu-intel64/README.txt#L22-L88 to setup my environment, but qemu report "Boot failed: Could not read from CDROM (code 0004)":
```
SeaBIOS (version 1.14.0-2)


iPXE (http://ipxe.org) 00:03.0 CA00 PCI2.10 PnP PMM+7FF8B3C0+7FECB3C0 CA00
                                                                               


Booting from Hard Disk...
Boot failed: could not read the boot disk

Booting from Floppy...
Boot failed: could not read the boot disk

Booting from DVD/CD...
Boot failed: Could not read from CDROM (code 0004)
Booting from ROM...
iPXE (PCI 00:03.0) starting execution...ok
iPXE initialising devices...ok



iPXE 1.0.0+git-20190125.36a4c85-5ubuntu4 -- Open Source Network Boot Firmware --
 http://ipxe.org
Features: DNS HTTP HTTPS iSCSI NFS TFTP AoE ELF MBOOT PXE bzImage Menu PXEXT

net0: 52:54:00:12:34:56 using 82540em on 0000:00:03.0 (open)
  [Link:up, TX:0 TXE:0 RX:0 RXE:0]
Configuring (net0 52:54:00:12:34:56)...... ok
net0: 10.0.2.15/255.255.255.0 gw 10.0.2.2
net0: fec0::5054:ff:fe12:3456/64 gw fe80::2
net0: fe80::5054:ff:fe12:3456/64
Nothing to boot: No such file or directory (http://ipxe.org/2d03e13b)
No more network devices

No bootable device.                      
```
Do you hit the similar issue? BTW, I use WSL2.

**Comment by masayuki2009** [comment]
>Do you hit the similar issue? BTW, I use WSL2.

@xiaoxiang781216 
Hmm, I've never tried the qemu-intel64 configuration on WSL2.

However, I know that the qemu-intel64 does not work on VMWare.
So I usually use a dedicated Ubuntu + Core i7 machine for it.

**Comment by xiaoxiang781216** [comment]
Ok, I will try the real device in Monday.

**Pull request merged**